src/ui/src/pages/AuthPage/AuthPage.tsx

```diff
@@ -1,22 +1,45 @@
-import React, { useState } from 'react';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useMutation } from 'react-query';
+import { toast, ToastContainer } from 'react-toastify';
 import RegistrationForm from '../../components/forms/RegistrationForm';
 import LoginForm from '../../components/forms/LoginForm';
 import { IRegisterFormValues } from '../../interfaces/Forms/IRegisterFormValues';
 import { ILoginFormValues } from '../../interfaces/Forms/ILoginFormValues';
+import { login, register } from '../../api/api';
+import setAuthenticationCookies from './Logic/setAuthenticationCookies';
 
 function AuthPage() {
   const [showLoginForm, setShowLoginForm] = useState<boolean>(false);
-
+  const navigate = useNavigate();
   const toggleForm = () => {
     setShowLoginForm((prev) => !prev);
   };
+  const loginMutation = useMutation(login);
+  const registrationMutation = useMutation(register);
+
+  const handleLogin = async (values: ILoginFormValues) => {
+    try {
+      const { token } = await loginMutation.mutateAsync(values);
+
+      setAuthenticationCookies(token);
 
-  const handleClickRegistration = (values: IRegisterFormValues) => {
-    console.log('Registration Form Values:', values);
+      navigate('/home');
+    } catch (error) {
+      toast.error('Login failed. Please check your credentials and try again.');
+    }
   };
 
-  const handleClickLogin = (values: ILoginFormValues) => {
-    console.log('Login Form Values:', values);
+  const handleRegistration = async (values: IRegisterFormValues) => {
+    try {
+      await registrationMutation.mutateAsync(values);
+      await handleLogin(values);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Turn it off on the application level or on the file level to be able to catch such errors without this dirty comments